src/coreComponents/physicsSolvers/SolverBase.cpp

```diff
@@ -494,162 +598,221 @@ real64 SolverBase::nonlinearImplicitStep( real64 const & time_n,
+        if( m_precond )
+        {
+          m_precond->clear();
+        }
 
-      // Compose parallel LA matrix/rhs out of local LA matrix/rhs
-      m_matrix.create( m_localMatrix.toViewConst(), m_dofManager.numLocalDofs(), MPI_COMM_GEOSX );
+        // Compose parallel LA matrix/rhs out of local LA matrix/rhs
+        m_matrix.create( m_localMatrix.toViewConst(), m_dofManager.numLocalDofs(), MPI_COMM_GEOSX );
 
-      // Output the linear system matrix/rhs for debugging purposes
-      debugOutputSystem( time_n, cycleNumber, newtonIter, m_matrix, m_rhs );
+        // Output the linear system matrix/rhs for debugging purposes
+        debugOutputSystem( time_n, cycleNumber, newtonIter, m_matrix, m_rhs );
 
-      // Solve the linear system
-      solveSystem( m_dofManager, m_matrix, m_rhs, m_solution );
+        // Solve the linear system
+        solveSystem( m_dofManager, m_matrix, m_rhs, m_solution );
 
-      // Output the linear system solution for debugging purposes
-      debugOutputSolution( time_n, cycleNumber, newtonIter, m_solution );
+        // Output the linear system solution for debugging purposes
+        debugOutputSolution( time_n, cycleNumber, newtonIter, m_solution );
 
-      scaleFactor = scalingForSystemSolution( domain, m_dofManager, m_solution.values() );
+        scaleFactor = scalingForSystemSolution( domain, m_dofManager, m_solution.values() );
 
-      if( !checkSystemSolution( domain, m_dofManager, m_solution.values(), scaleFactor ) )
-      {
-        // TODO try chopping (similar to line search)
-        GEOSX_LOG_RANK_0( "    Solution check failed. Newton loop terminated." );
-        break;
-      }
+        if( !checkSystemSolution( domain, m_dofManager, m_solution.values(), scaleFactor ) )
+        {
+          // TODO try chopping (similar to line search)
+          GEOSX_LOG_RANK_0( "    Solution check failed. Newton loop terminated." );
+          break;
+        }
 
-      // apply the system solution to the fields/variables
-      applySystemSolution( m_dofManager, m_solution.values(), scaleFactor, domain );
+        // apply the system solution to the fields/variables
+        applySystemSolution( m_dofManager, m_solution.values(), scaleFactor, domain );
 
-      // update non-primary variables (constitutive models)
-      updateState( domain );
+        // update non-primary variables (constitutive models)
+        updateState( domain );
 
-      lastResidual = residualNorm;
-    }
+        lastResidual = residualNorm;
+      } // end of Newton's loop
 
-    if( isConverged )
+      if( isNewtonConverged )
+      {
+        isConfigurationLoopConverged = updateConfiguration( domain );
+
+        if( isConfigurationLoopConverged )
+        {
+          break; // get out of configuration loop coz everything converged.
+        }
+        else
+        {
+          GEOSX_LOG_LEVEL_RANK_0( 1, "  --------Configuration did not converge. Testing new configuration." );
```

Comment:
I've opened a whole issue about this, our logging is very inconsistent (indentation, padding characters, verbosity levels). The configuration loop will be the latest addition :) This is general comment, not something you should try to fix in this PR.